src/runtime/generator.cpp

```diff
@@ -88,6 +111,7 @@ void generatorEntry(BoxedGenerator* g) {
     // we returned from the body of the generator. next/send/throw will notify the caller
     g->entryExited = true;
     threading::popGenerator();
+    freeGeneratorStack(g);
```

Comment:
Honestly I was kinda surprised this worked, but I suppose the register allocator must be stashing `g` in a callee-save register before the call to `freeGeneratorStack`.

It'll work to put it on the other side (and will be safer).  Given my limited understanding of python's iterator protocol, it seems like the only downside would be that the stack could stick around a little longer.